cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -505,23 +514,40 @@ def _check_file_systems(self, are_all_security_groups_customized, file_systems,
                     network_interface_responses.append(network_interfaces_data[network_interface_id])
 
                 network_interfaces = [ni for ni in network_interface_responses if ni.get("VpcId") == vpc_id]
-                ports = FSX_PORTS[file_system.file_system_type]
-                for port in ports:
-                    fs_access = False
-                    for network_interface in network_interfaces:
-                        # Get list of security group IDs
-                        sg_ids = [sg.get("GroupId") for sg in network_interface.get("Groups")]
-                        if _check_in_out_access(sg_ids, port=port, is_cidr_optional=are_all_security_groups_customized):
-                            fs_access = True
-                            break
-                    if not fs_access:
+
+                for protocol, ports in FSX_PORTS[file_system.file_system_type].items():
+                    missing_ports = self._get_missing_ports(
+                        are_all_security_groups_customized, network_interfaces, ports, protocol
+                    )
+
+                    if missing_ports:
                         self._add_failure(
                             f"The current security group settings on file system '{file_system_id}' does not"
                             " satisfy mounting requirement. The file system must be associated to a security group"
-                            f" that allows inbound and outbound TCP traffic through ports {ports}.",
+                            f" that allows inbound and outbound {protocol.upper()} traffic through ports {ports}. "
+                            f"Missing ports: {missing_ports}",
                             FailureLevel.ERROR,
                         )
 
+    def _get_missing_ports(self, are_all_security_groups_customized, network_interfaces, ports, protocol):
+        missing_ports = []
+        for port in ports:
+            fs_access = False
+            for network_interface in network_interfaces:
+                # Get list of security group IDs
+                sg_ids = [sg.get("GroupId") for sg in network_interface.get("Groups")]
+                if _check_in_out_access(
+                    sg_ids,
+                    port=port,
+                    is_cidr_optional=are_all_security_groups_customized,
```

Comment:
If all security groups are customized, CIDR specification is not required. Because custom security groups can allow traffics by specifying security group id of themselves or others.
On the contrary, if cluster nodes use ParallelCluster managed security group, the security groups of the file system has to have CIDR specification to enable traffics from the file system to the nodes